docs/what-is-hls.md

```diff
@@ -0,0 +1,58 @@
+HLS provides [many](./features.md) (but not all) of the features that the LSP protocol supports.
+
+But HLS only provides the server part of the setup.
+In order to actually use it you also need a client (editor).
+The client is responsible for managing your interaction with the server: launching it, dispatching commands to it, and displaying or implementing responses.
+Some clients will even install the server binaries for you!
+
+Common clients include:
+- VSCode (the reference implementation for a LSP client)
+- Emacs, with the `lsp-mode`+`lsp-haskell` or `eglot` packages
+- Vim/neovim, with the `coc.vim` or `LanguageClient-neovim` packages
+- Kate
+- ... and more every day!
+
+### LSP terminology
+
+Here are a few pieces of jargon that you may come across in the HLS docs or when discussing problems:
+
+- *Code action*: A code action is a specific action triggered by a user on a particular region of code. Examples might include "add a type signature to this function".
+- *Code lens*: A pre-rendered edit or action shown in the body of the document itself, usually triggered with a click. Examples might include "the type signature for a function, which is actually inserted on click".
+- *Completion item*: An item that can be inserted into the text, including its metadata.
+- *Diagnostic*: Any information about the project that is shown in the editor, including errors, warnings, and hints from tools such as hlint.
+- *Semantic highlighting*: Special syntax highlighting performed by the server.
+
+## The structure of HLS
+
+### HLS and its wrapper
+
+HLS is a binary that must be compiled with the same version of GHC as the project you are using it on.
+For this reason it is usually distributed as a _collection_ of binaries, along with a `haskell-language-server-wrapper` executable that selects the correct one based on which version of GHC it thinks you are using.
+
+In general you can use `haskell-language-server-wrapper` wherever you need to run `haskell-language-server`.
+
+### HLS plugins
+
+HLS has a plugin architecture, whereby individual pieces of functionality are provided by smaller packages that just do one thing.
+Plugins can also be disabled independently to allow users to customize the behaviour of HLS to their liking.
+
+These plugins all (currently) live in the HLS repository and are developed in tandem with the core HLS functionality.
+
+### hie-bios
+
+HLS needs to know how to build your Haskell project: what flags to pass, what packages to provide, etc.
+It gets this information from the build system used by your project (typically `cabal` or `stack`).
+The tool used to do this is called [`hie-bios`](https://github.com/haskell/hie-bios).
+`hie-bios` calls the strategy it uses to get compilation flags (e.g. "ask `cabal`") a "cradle".
```

Comment:
I would mention informally here what is a "cradle", we are using that word continuously and it is used in the new troubleshooting.

Also would be really great describe the implicit-hie discovery process (not sure if mentioning the library). But that can be left for a follow up

Reply:
Yeah, although I'd welcome some contributions here. I'm not sure _I_ really know what goes on...

Reply:
hmm, err, well, what is really a cradle? @fendor? 😆